src/plugins/muc-views/utils.js

```diff
@@ -127,9 +127,24 @@ export function getAutoCompleteListItem (text, input) {
     return element;
 }
 
-export async function getAutoCompleteList () {
-    const models = [...(await api.rooms.get()), ...(await api.contacts.get())];
-    const jids = [...new Set(models.map(o => Strophe.getDomainFromJid(o.get('jid'))))];
+// models array is globalized to introduce caching of the list from https://search.jabber.network/rooms/1.
+const models = [];
+let callFetchListOfRooms = false;
+
+async function fetchListOfRooms() {
+    const response = await fetch('https://search.jabber.network/api/1.0/rooms');
+    const data = await response.json();
+    for (var x = 0; x < data.items.length; x++) {
+        models.push(data.items[x]["address"]);
+    }
+}
+
+export function getAutoCompleteList() {
+    if (callFetchListOfRooms === false) {
+        fetchListOfRooms();
+        callFetchListOfRooms = true;
+    }
+    const jids = [...new Set(models)];
```

Comment:
Instead of always making sure that we return unique jids, we can rather just do that once in `fetchListOfRooms` when we populate `models`.